pkg/operator/cluster/ceph/osd/pod.go

```diff
@@ -61,6 +61,152 @@ func (c *Cluster) makeDaemonSet(selection rookalpha.Selection, config rookalpha.
 	}
 }
 
+func (c *Cluster) makePod(nodeName string, devices []rookalpha.Device, selection rookalpha.Selection, resources v1.ResourceRequirements, config rookalpha.Config) *v1.Pod {
```

Comment:
i think we only need one code path for creating an osd deployment. The `makePod` and `makeOSDReplicaSet` methods seem like they are doing the same thing?